packages/jsii-rosetta/lib/commands/transliterate.ts

```diff
@@ -0,0 +1,172 @@
+import { Assembly, Docs, SPEC_FILE_NAME, Type, TypeKind } from '@jsii/spec';
+import { readJson, writeJson } from 'fs-extra';
+import { resolve } from 'path';
+
+import { TargetLanguage } from '../languages';
+import { Rosetta } from '../rosetta';
+import { typeScriptSnippetFromSource } from '../snippet';
+import { Translation } from '../tablets/tablets';
+
+export interface TransliterateAssemblyOptions {
+  readonly strict?: boolean;
+}
+
+/**
+ * Prepares transliterated versions of the designated assemblies into the
+ * selected taregt languages.
+ *
+ * @param assemblyLocations the directories which contain assemblies to
+ *                          transliterate.
+ * @param targetLanguages   the languages into which to transliterate.
+ * @param tabletLocation    an optional Rosetta tablet file to source
+ *                          pre-transliterated snippets from.
+ *
+ * @experimental
+ */
+export async function transliterateAssembly(
+  assemblyLocations: readonly string[],
+  targetLanguages: readonly TargetLanguage[],
+  tabletLocation?: string,
+  options: TransliterateAssemblyOptions = {},
+): Promise<void> {
+  const rosetta = new Rosetta({ liveConversion: true, targetLanguages });
+  if (tabletLocation) {
+    await rosetta.loadTabletFromFile(tabletLocation);
+  }
+  const assemblies = await Promise.all(
```

Comment:
Style: the functional style is very neat but TBH makes the logic pretty hard to read (at least for me). I think imperative style will be more readable and easier to maintain long-term.